src/application-manager.d.ts

```diff
@@ -51,7 +51,9 @@ export class ApplicationManager extends EventEmitter {
 		opts: Options,
 	): Bluebird<void>;
 
-	public getStatus(): Promise<DeviceApplicationState>;
+	// FIXME: Type this properly as it's some mutant state between
```

Comment:
What was wrong with the previous typing?

Reply:
It was correct for `DeviceApplicationState` in the form before this PR, but that interface was wrong. Basically this type is a mix between the state endpoint for a single application and the internal of ApplicationManager. I was going to add the other type to the d.ts but I'm going to be rewriting ApplicaitonManager in typescript very soon, so I'd rather do it then. 

I can add the typing to this PR instead if you prefer?